### PR TITLE
2016.09.07 update

### PR DESCRIPTION
**Pull request by steenzout**
- added integration tests
- added more unit tests (and a few stub methods)
- change code modules, classes, etc to be more Python-like
- added docstring

- `bucket` functions for the HTTP client pass integration tests

I'll keep working on this but this PR would be useful to merge to have future contributions (by others) based on this branch.

### CODE REVIEW

**Comment by Miskerest** [comment]
Should I go ahead and merge despite Travis and coveralls warnings?

**Comment by steenzout** [comment]
@Miskerest I would say so.

the travis-ci is failing on Python 3.4 and 3.3 but working on 3.5.
py27 and pypy also pass.

considering the amount of code added it was expected coveralls to complain (lower coverage).

I'm tackling that but it will take a bit of time :)

File: storj/http.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+            keyfile.write(self.public_key.to_pem())
+
+        print("Writing private key to file... Keep this secret!")
+        with open('private.pem', 'wb') as keyfile:
+            keyfile.write(self.private_key.to_pem())
+
+        print("Wrote keyfiles to dir: " + os.getcwd())
+
+    def key_generate(self):
+        print("This will replace your public and private keys in 3 seconds...")
```
Comment (super3) [inline storj/http.py]: Why wait for public and private key generation?

**Comment by Miskerest** [comment]
@super3 To give the user a chance to cancel in case they don't want their keys deleted.

**Comment by super3** [comment]
@Miskerest Probably should just give them a prompt then.

**Pull request merged**